iceoryx_utils/include/iceoryx_utils/cxx/sized_uninitialized_array.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
```

Comment:
I would rename the class into `uninitialized_array` then it follows more the C++/our conventions.

Reply:
Sure, can change that. 
I tried to follow the naming conventions described in CONTRIBUTING.md, which suggest to use camel case for class names.